src/PHPMailer.php

```diff
@@ -3857,6 +3857,10 @@ protected function serverHostname()
      */
     public static function isValidHost($host)
     {
+        if (version_compare(PHP_VERSION, '7.0.0') >= 0) {
+            return is_scalar($host) && filter_var($host, FILTER_VALIDATE_DOMAIN) !== false;
```

Comment:
`is_scalar` returns true for any string, but `FILTER_VALIDATE_DOMAIN` has quite a few problems, for example [this bug report](https://bugs.php.net/bug.php?id=77331). It also allows things that are definitely not usable hostnames, like `192.168.0.1/x` and `example`, even when the `FILTER_FLAG_HOSTNAME` is added for the latter example, but if you do add that flag it rejects IPv6 addresses. This isn't looking very reliable!

Reply:
I added the "is_scalar", so filter_var isn't fed with a class/array/etc.

Didn't knew about the unreliability of  FILTER_VALIDATE_DOMAIN/FILTER_FLAG_HOSTNAME .... "Great"

Reply:
Removed usage of FILTER_VALIDATE_DOMAIN/FILTER_FLAG_HOSTNAME